app/lib/common/providers/chat_providers.dart

```diff
@@ -125,3 +129,49 @@ final memberProvider =
   }
   return await convo.getMember(userId);
 });
+
+final imageFileFromMessageIdProvider =
+    FutureProvider.family<File, String>((ref, messageId) async {
+  final convo = await ref.read(currentConvoProvider.future);
+  //Check if video file is available
+  final tempDir = await getTemporaryDirectory();
+  var imageFile = File('${tempDir.path}/image-$messageId.jpg');
+  var isImageAvailable = await imageFile.exists();
+
+  if (!isImageAvailable) {
+    if (convo != null) {
+      //If video file is not available on local store then save it
+      AsyncValue<Uint8List> videoData = AsyncValue.data(
+        await convo.imageBinary(messageId).then((value) => value.asTypedList()),
+      );
+      if (videoData.asData != null) {
```

Comment:
We should probably handle error case here like some error string specifying that media can't be opened or unavailable.